kafka-client/src/main/java/dev/responsive/kafka/config/ResponsiveConfig.java

```diff
@@ -190,4 +211,57 @@ public String toString() {
   public ResponsiveConfig(final Map<?, ?> originals) {
     super(CONFIG_DEF, originals);
   }
+
+  public SubPartitioner getSubPartitioner(
+      final CassandraClient cassandraClient,
+      final Admin admin,
+      final TableName name,
+      final String changelogTopicName
+  ) {
+    final OptionalInt actualRemoteCount = cassandraClient.numPartitions(name.cassandraName());
+    final int kafkaPartitions;
+    try {
+      kafkaPartitions = admin.describeTopics(List.of(changelogTopicName))
+          .allTopicNames()
+          .get()
+          .get(changelogTopicName)
+          .partitions()
+          .size();
+    } catch (InterruptedException | ExecutionException e) {
+      throw new RuntimeException(e);
+    }
+
+    return getSubPartitioner(
+        actualRemoteCount,
+        kafkaPartitions,
+        getInt(STORAGE_DESIRED_NUM_PARTITION_CONFIG),
+        name,
+        changelogTopicName
+    );
+  }
+
+  static SubPartitioner getSubPartitioner(
```

Comment:
This should be private, right? If it's only visible for testing, we can just go through the public method above with a mock admin right?

(Or make this a static util method in the `SubPartitioner` class instead)